### PR TITLE
Error in MinCovDet covariance estimator with sklearn 1.6.0

### PR DESCRIPTION
**Pull request by gabelstein**
Not sure why this happened, but kept throwing errors all around.
I have no idea how to fix the last error, though.

### CODE REVIEW

**Comment by gabelstein** [comment]
Seems to be a problem with sklearn 1.6.0

**Comment by agramfort** [comment]
sklearn 1.6 was released yesterday. this explains the sudden breakage.

**Comment by gabelstein** [comment]
Ah, yes. 
The error seems to be in the changes applied to the _c_step function of sklearn/covariance/_robust_covariance.py
No idea why it doesn't work, though.
But seems to be an sklearn problem, not pyriemann.

maybe fix sklearn version vor now to <1.6?

File: pyriemann/classification.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def _mode_2d(X, axis=1):
     return mode
 
 
-class MDM(BaseEstimator, ClassifierMixin, TransformerMixin):
```
Comment (qbarthelemy) [inline pyriemann/classification.py:31]: If you remove `TransformerMixin`, `fit_transform()` is no longer defined.

File: tests/test_classification.py [hunk 2]
```diff
@@ -458,3 +458,12 @@ def test_class_distinctiveness(n_classes, metric_mean, metric_dist, exponent,
     assert isinstance(class_dis, float), "Unexpected object of class_dis"
     assert isinstance(num, float), "Unexpected object of num"
     assert isinstance(denom, float), "Unexpected object of denum"
+
+
+@pytest.mark.parametrize("classif", [MDM, FgMDM, MeanField])
```
Comment (qbarthelemy) [inline tests/test_classification.py:463]: My next PR was to add this test ;-)

To follow the framework of current tests, can you add at line 127 this function?
```
def clf_fittransform(classif, mats, labels):
    clf = classif()
    transf = clf.fit_transform(mats, labels)
    transf2 = clf.fit(mats, labels).transform(mats)
    assert_array_equal(transf, transf2)
```
and then, update line 78 with:
```
    if classif in (MDM, FgMDM, MeanField):
        clf_transform(classif, mats, labels)
        clf_fittransform(classif, mats, labels)
```

**Comment by qbarthelemy** [comment]
@anntzer, the latest release of sklearn broke our tests, especially on `fast_mcd`.
It seems to be related to https://github.com/scikit-learn/scikit-learn/pull/29835
Would you agree to help us understand the problem?

```
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/pyriemann/utils/covariance.py:68: in _mcd
    _, C, _, _ = fast_mcd(X.T, **kwds)
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/sklearn/covariance/_robust_covariance.py:555: in fast_mcd
    locations_best, covariances_best, _, _ = select_candidates(
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/sklearn/covariance/_robust_covariance.py:322: in select_candidates
    _c_step(
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/sklearn/covariance/_robust_covariance.py:161: in _c_step
    support_indices = np.argpartition(dist, n_support - 1)[:n_support]
/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/numpy/_core/fromnumeric.py:962: in argpartition
    return _wrapfunc(a, 'argpartition', kth, axis=axis, kind=kind, order=order)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

obj = array([19., 19., 19., 19., 19., 19., 19., 19., 19., 19., 19., 19., 19.,
       19., 19., 19., 19., 19., 19., 19.])
method = 'argpartition', args = (20,)
kwds = {'axis': -1, 'kind': 'introselect', 'order': None}
bound = <built-in method argpartition of numpy.ndarray object at 0x7ff7c2837c30>

    def _wrapfunc(obj, method, *args, **kwds):
        bound = getattr(obj, method, None)
        if bound is None:
            return _wrapit(obj, method, *args, **kwds)
    
        try:
>           return bound(*args, **kwds)
E           ValueError: kth(=20) out of bounds (20)

/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/numpy/_core/fromnumeric.py:57: ValueError
```
See complete log here: https://github.com/pyRiemann/pyRiemann/actions/runs/12277182896/job/34256013254

**Comment by anntzer** [comment]
See https://github.com/scikit-learn/scikit-learn/pull/30483.

**Comment by qbarthelemy** [comment]
While waiting for the problem to be solved, I propose to pass the error and move forward because the pull-requests to merge are accumulating. We will remove the skip once sklearn 1.6.1 is released.

**Pull request merged**